diff --git a/src/HttpResponse.cc b/src/HttpResponse.cc
index 500cd64..a7b0e7c 100644
--- a/src/HttpResponse.cc
+++ b/src/HttpResponse.cc
@@ -165,7 +165,7 @@ void HttpResponse::processRedirect()
         cuid_, req->getCurrentUri().c_str()));
   }
 
-  A2_LOG_NOTICE(fmt(MSG_REDIRECT, cuid_,
+  A2_LOG_INFO(fmt(MSG_REDIRECT, cuid_,
                     httpRequest_->getRequest()->getCurrentUri().c_str()));
 }
 
