diff --git a/browser/base/content/browser-sets.inc b/browser/base/content/browser-sets.inc
index 11464065b..e0765598a 100644
--- a/browser/base/content/browser-sets.inc
+++ b/browser/base/content/browser-sets.inc
@@ -206,7 +206,7 @@
     <key keycode="VK_F11" command="View:FullScreen"/>
 #endif
 #ifndef XP_WIN
-    <key id="key_toggleReaderMode" key="&toggleReaderMode.key;" command="View:ReaderView" modifiers="accel,alt" disabled="true"/>
+    <key id="key_toggleReaderMode" key="&toggleReaderMode.key;" command="View:ReaderView" modifiers="alt"/>
 #else
     <key id="key_toggleReaderMode" keycode="&toggleReaderMode.win.keycode;" command="View:ReaderView" disabled="true"/>
 #endif
