diff --git a/services/localzone.c b/services/localzone.c
index 54f55ab..1d0cad2 100644
--- a/services/localzone.c
+++ b/services/localzone.c
@@ -1258,7 +1258,11 @@ local_encode(struct query_info* qinfo, struct module_env* env,
 	uint16_t udpsize;
 	/* make answer with time=0 for fixed TTL values */
 	memset(&rep, 0, sizeof(rep));
-	rep.flags = (uint16_t)((BIT_QR | BIT_AA | BIT_RA) | rcode);
+	/* local patch: use dnstap_send_identity to return AA bit instead of RA */
+	if(env->cfg->dnstap_send_identity) {
+		rep.authoritative = 1;
+		rep.flags = (uint16_t)((BIT_QR | BIT_AA) | rcode); }
+	else rep.flags = (uint16_t)((BIT_QR | BIT_AA | BIT_RA) | rcode);
 	rep.qdcount = 1;
 	if(ansec)
 		rep.an_numrrsets = 1;
