--- weavedlibrary.bak	2017-04-05 14:27:24.428174984 +0500
+++ weavedlibrary	2017-04-05 15:21:51.482242398 +0500
@@ -73,6 +73,7 @@
 #### check for updated installer
 checkForUpdate()
 {
+    return # disabled on ArchLinuxARM
     echo
     echo "-------------------"
     echo    
@@ -1023,6 +1024,7 @@
 ######### Add line to cron ######
 cronAddLine()
 {
+    return 0
     crontab -l 2>/dev/null 1> "$TMP_DIR"/.crontab_old
     echo "$1" >> "$TMP_DIR"/.crontab_old
     logger "cronAddline $1"
@@ -1033,6 +1035,7 @@
 ######### Disable Weaved services to start at reboot time ######
 cronRemoveLine()
 {
+    return 0
     crontab -l | grep -v "$1" | cat > $TMP_DIR/.crontmp
     crontab $TMP_DIR/.crontmp
 }
@@ -1040,14 +1043,50 @@
 ######### Enable remot3.it services to start at reboot time ######
 enableStartup()
 {
-    cronAddLine "@reboot $BIN_DIR/$STARTEMUP"
+	state=$(systemctl is-enabled weaved 2>/dev/null)
+	[[ -n "$state" ]] || {
+		cat >/etc/systemd/system/weaved.service <<'EOF'
+[Service]
+RemainAfterExit=yes
+ExecStart=/usr/bin/weavedstart.sh
+ExecStop=/usr/bin/weavedstop.sh
+
+[Install]
+WantedBy=multi-user.target
+EOF
+		cat >/etc/systemd/system/weaved-reboot.service <<'EOF'
+[Unit]
+Before=weaved.service
+ConditionPathExists=/root/.remot3.it/reboot.sh
+
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/bash /root/.remot3.it/reboot.sh
+
+[Install]
+WantedBy=multi-user.target
+EOF
+	}
+	[[ "$state" = enabled ]] || {
+		systemctl daemon-reload
+		systemctl enable weaved
+		systemctl enable weaved-reboot
+	}
 }
 ######### End Enable remot3.it services to start at reboot time ######
 
 ######### Disable remot3.it services to start at reboot time ######
 disableStartup()
 {
-    cronRemoveLine "@reboot $BIN_DIR/$STARTEMUP"
+	state=$(systemctl is-enabled weaved 2>/dev/null)
+	[[ "$state" != enabled ]] || {
+		systemctl disable weaved
+		systemctl disable weaved-reboot
+	}
+	[[ -z "$state" ]] || {
+		rm -f /etc/systemd/system/weaved{,-reboot}.service
+		systemctl daemon-reload
+	}
 }
 
 ######### Install Start/Stop Scripts #########
@@ -1067,7 +1106,8 @@
             cp "$WEAVED_LIB_DIR"/scripts/$STARTEMUP "$BIN_DIR"
 	    #      printf "$STARTEMUP copied to %s\n" "$BIN_DIR"
 	fi
-	checkCron=$(crontab -l 2> /dev/null | grep $STARTEMUP | wc -l)
+	checkCron=$(systemctl is-enabled weaved 2>/dev/null)
+	[[ "$checkCron" = enabled ]] && checkCron=1 || checkCron=0
 	debug "CheckCron: $checkCron\n"
 	if [[ $checkCron == 0 && "$STARTDAEMONSONREBOOT" == "1" ]]; then
 	    enableStartup
