diff --git a/dom/base/Document.cpp b/dom/base/Document.cpp
index 0b5db538..5e637568 100644
--- a/dom/base/Document.cpp
+++ b/dom/base/Document.cpp
@@ -17818,9 +17818,9 @@ ColorScheme Document::DefaultColorScheme() const {
 }
 
 ColorScheme Document::PreferredColorScheme(IgnoreRFP aIgnoreRFP) const {
-  if (ShouldResistFingerprinting() && aIgnoreRFP == IgnoreRFP::No) {
-    return ColorScheme::Light;
-  }
+  // if (ShouldResistFingerprinting() && aIgnoreRFP == IgnoreRFP::No) {
+  //   return ColorScheme::Light;
+  // }
 
   if (nsPresContext* pc = GetPresContext()) {
     if (auto scheme = pc->GetOverriddenOrEmbedderColorScheme()) {
