diff --git a/dom/base/Document.cpp b/dom/base/Document.cpp
index 55457d6..b594ec1 100644
--- a/dom/base/Document.cpp
+++ b/dom/base/Document.cpp
@@ -18446,10 +18446,10 @@ ColorScheme Document::DefaultColorScheme() const {
 }
 
 ColorScheme Document::PreferredColorScheme(IgnoreRFP aIgnoreRFP) const {
-  if (ShouldResistFingerprinting(RFPTarget::CSSPrefersColorScheme) &&
-      aIgnoreRFP == IgnoreRFP::No) {
-    return ColorScheme::Light;
-  }
+  // if (ShouldResistFingerprinting(RFPTarget::CSSPrefersColorScheme) &&
+  //     aIgnoreRFP == IgnoreRFP::No) {
+  //   return ColorScheme::Light;
+  // }
 
   if (nsPresContext* pc = GetPresContext()) {
     if (auto scheme = pc->GetOverriddenOrEmbedderColorScheme()) {
