diff --git a/dom/base/Document.cpp b/dom/base/Document.cpp
index 3efc8c5..2d1c66b 100644
--- a/dom/base/Document.cpp
+++ b/dom/base/Document.cpp
@@ -17807,10 +17807,10 @@ ColorScheme Document::DefaultColorScheme() const {
 }
 
 ColorScheme Document::PreferredColorScheme(IgnoreRFP aIgnoreRFP) const {
-  if (aIgnoreRFP == IgnoreRFP::No &&
-      nsContentUtils::ShouldResistFingerprinting(this)) {
-    return ColorScheme::Light;
-  }
+  // if (aIgnoreRFP == IgnoreRFP::No &&
+  //     nsContentUtils::ShouldResistFingerprinting(this)) {
+  //   return ColorScheme::Light;
+  // }
 
   if (nsPresContext* pc = GetPresContext()) {
     if (auto scheme = pc->GetOverriddenOrEmbedderColorScheme()) {
