diff --git a/dom/base/Document.cpp b/dom/base/Document.cpp
index 132c0ecbfd..e00d669704 100644
--- a/dom/base/Document.cpp
+++ b/dom/base/Document.cpp
@@ -16355,10 +16355,10 @@ void Document::RemoveToplevelLoadingDocument(Document* aDoc) {
 
 StylePrefersColorScheme Document::PrefersColorScheme(
     IgnoreRFP aIgnoreRFP) const {
-  if (aIgnoreRFP == IgnoreRFP::No &&
-      nsContentUtils::ShouldResistFingerprinting(this)) {
-    return StylePrefersColorScheme::Light;
-  }
+  // if (aIgnoreRFP == IgnoreRFP::No &&
+  //     nsContentUtils::ShouldResistFingerprinting(this)) {
+  //   return StylePrefersColorScheme::Light;
+  // }
 
   if (nsPresContext* pc = GetPresContext()) {
     if (auto devtoolsOverride = pc->GetOverridePrefersColorScheme()) {
