diff --git a/servo/components/style/driver.rs b/servo/components/style/driver.rs
index eab7c3a6a3..9ab3e902c9 100644
--- a/servo/components/style/driver.rs
+++ b/servo/components/style/driver.rs
@@ -27,5 +27,5 @@ fn should_report_statistics() -> bool {
 #[cfg(feature = "gecko")]
 fn should_report_statistics() -> bool {
-    unsafe { crate::gecko_bindings::structs::ServoTraversalStatistics_sActive }
+    false
 }
 
